lib/brainstem/api_docs/formatters/open_api_specification/version_2/endpoint/response_definitions_formatter.rb

```diff
@@ -73,41 +73,71 @@ def success_response_description
               def format_schema_response!
                 return if presenter.nil?
 
-                brainstem_key = presenter.brainstem_keys.first
-                model_klass   = presenter.target_class
-
                 output.merge! '200' => {
                   description: success_response_description,
                   schema: {
+                    type: 'object',
+                    properties: properties
+                  }
+                }
+              end
+
+              def properties
+                brainstem_key = presenter.brainstem_keys.first
+                model_klass = presenter.target_class
+
+                {
+                  count: type_and_format('integer'),
+                  meta: {
                     type: 'object',
                     properties: {
                       count: type_and_format('integer'),
-                      meta: {
-                        type: 'object',
-                        properties: {
-                          count:       type_and_format('integer'),
-                          page_count:  type_and_format('integer'),
-                          page_number: type_and_format('integer'),
-                          page_size:   type_and_format('integer'),
-                        }
-                      },
-                      results: {
-                        type: 'array',
-                        items: {
-                          type: 'object',
-                          properties: {
-                            key: type_and_format('string'),
-                            id:  type_and_format('string')
-                          }
-                        }
-                      },
-                      brainstem_key => {
-                        type: 'object',
-                        additionalProperties: {
-                          '$ref' => "#/definitions/#{model_klass}"
-                        }
+                      page_count: type_and_format('integer'),
+                      page_number: type_and_format('integer'),
+                      page_size: type_and_format('integer'),
+                    }
+                  },
+                  results: {
+                    type: 'array',
+                    items: {
+                      type: 'object',
+                      properties: {
+                        key: type_and_format('string'),
+                        id: type_and_format('string')
                       }
                     }
+                  },
+                  brainstem_key => {
```

Comment:
I think you can pull this into a method where you pass in the key and target_klass

Reply:
👍